collector/collector.go

```diff
@@ -3,161 +3,109 @@ package collector
-	m := struct {
-		Result interface{} `json:"result"`
-		Error  string      `json:"error"`
-	}{
-		Result: ret,
+func CastMapToStruct(m interface{}, ret interface{}) error {
+	data, err := json.Marshal(m)
+	if err != nil {
+		return fmt.Errorf("failed to marshal body interface{}: %w", err)
 	}
-	decoder := json.NewDecoder(resp.Body)
-	err := decoder.Decode(&m)
+	err = json.Unmarshal(data, ret)
 	if err != nil {
 		return fmt.Errorf("failed to unmarshal response body: %w", err)
 	}
-	if m.Error != "" {
-		return fmt.Errorf(m.Error)
-	}
 	return nil
 }
 
 type Collector interface {
-	Get(metric Metrics) (float64, error)
+	Get(metric Metric) (float64, error)
+	ProvidedMetrics() []Metric
 	UpdateData(ctx context.Context) error
 }
 
 type HttpClient interface {
 	Do(req *http.Request) (*http.Response, error)
 }
 
-func NewLCDCollector(endpoint string, RewardContractAddress string, logger *logrus.Logger) LCDCollector {
+func NewLCDCollector(logger *logrus.Logger) LCDCollector {
 	return LCDCollector{
-		logger:                logger,
-		LCDEndpoint:           endpoint,
-		RewardContractAddress: RewardContractAddress,
-		HttpClient:            http.DefaultClient,
+		Metrics:   make(map[Metric]Monitor),
+		logger:    logger,
+		apiClient: client.NewHTTPClient(nil),
 	}
 }
 
 type LCDCollector struct {
-	logger                *logrus.Logger
-	LCDEndpoint           string
-	HubAddress            string
-	RewardContractAddress string
-	BlunaContractAddress  string
-	CollectedData         CollectedData
-	HttpClient            HttpClient
+	Metrics   map[Metric]Monitor
+	Monitors  []Monitor
+	logger    *logrus.Logger
+	apiClient *client.TerraLiteForTerra
 }
 
-func (c LCDCollector) Get(metric Metrics) (float64, error) {
-	switch metric {
-	case BlunaBondedAmount:
-		return strconv.ParseFloat(c.CollectedData.HubState.TotalBondAmount, 64)
-	case GlobalIndex:
-		return strconv.ParseFloat(c.CollectedData.RewardState.GlobalIndex, 64)
-	case BlunaTotalSupply:
-		return strconv.ParseFloat(c.CollectedData.BlunaTokenInfo.TotalSupply, 64)
+func (c LCDCollector) ProvidedMetrics() []Metric {
+	metrics := []Metric{}
+	for m := range c.Metrics {
+		metrics = append(metrics, m)
 	}
-	return 0, fmt.Errorf("metric \"%s\" not found", metric)
+	return metrics
 }
 
-func (c LCDCollector) buildRequest(ctx context.Context, contractAddress string, query interface{}) (*http.Request, error) {
-	url := fmt.Sprintf("%s/wasm/contracts/%s/store", c.LCDEndpoint, contractAddress)
-	req, err := http.NewRequestWithContext(ctx, "GET", url, nil)
-	if err != nil {
-		return nil, fmt.Errorf("failed to prapare request for %+v query: %w", query, err)
-	}
-	q := req.URL.Query()
-	queryRaw, err := json.Marshal(query)
-	if err != nil {
-		return nil, fmt.Errorf("failed to marshal %+v query: %w", query, err)
-	}
-	q.Add("query_msg", string(queryRaw))
-	req.URL.RawQuery = q.Encode()
-	return req, nil
+func (c *LCDCollector) SetTransport(transport runtime.ClientTransport) {
```

Comment:
Is SetTransport used only in tests? If so, it should be made private and a comment should be added.

Reply:
removed method entirely